R/primary-coding-suep.R

```diff
@@ -559,22 +579,22 @@ build_who_scale_suep_df <- function(trial_data, pid, docid, visitid){
                                       .data$gec_oxy_start_d.factor == "Auf Woche genau" ~ floor_date(.data$gec_oxy_start.date, unit = "week", week_start = getOption ("lubridate.week.start", 1)),
                                       TRUE ~ .data$gec_oxy_start.date),
       ecu_oxy_interval = interval(.data$gec_oxy_start.date, .data$gec_oxy_end.date),
-      gec_oxy_type.factor = factor(fct_reorder(.data$gec_oxy_type.factor, .data$gec_oxy_type), ordered = TRUE)) %>%
-    select(pid, .data$gec_oxy, .data$gec_oxy_type.factor, .data$gec_oxy_start.date, .data$gec_oxy_start_d.factor, .data$gec_oxy_start_uk.factor, 
+      gec_oxy_type.factor = factor(fct_reorder(.data$gec_oxy_type.factor, .data$gec_oxy_type, na.rm = FALSE), ordered = TRUE)) %>%
+    select(pid, .data$gec_oxy.factor, .data$gec_oxy_type, .data$gec_oxy_type.factor, .data$gec_oxy_start.date, .data$gec_oxy_start_d.factor, .data$gec_oxy_start_uk.factor, 
            .data$gec_oxy_end.date, .data$gec_oxy_end_d.factor, .data$gec_oxy_end_uk.factor, .data$gec_oxy_end_on.factor, .data$treatment_update.date, 
            .data$ecu_oxy_interval) #%>%
   #group_by(!!sym(pid), .data$gec_oxy) # %>%
   # nest(oxy_data = -c(!!sym(pid), .data$gec_oxy)) 
   
   
-  # Hospitalisation (one row per pat per start date) --------------------------------------------------------
+  # Hospitalisation (one row per pat per start date) ---------------------------
   
   hospital_data <- trial_data$eward %>%
     bind_rows(trial_data$eresid) %>% 
     bind_rows(treatment_update_date) %>%
```

Comment:
there might be a big gap between last update date and export date
we decided on update date, because this is the date we save know the therapies/symptoms were still present when its marked as "andauernd"